src/db/boxStore.js

```diff
@@ -198,20 +198,26 @@ module.exports.buyIn = async (boxBarcode, boxCount) => {
             .leftJoin('PRICE', 'RVBOX.itembarcode', 'PRICE.barcode')
             .leftJoin('RVITEM', 'PRICE.itemid', 'RVITEM.itemid')
             .where('RVBOX.barcode', boxBarcode)
-            .first('RVBOX.itemcount', 'PRICE.priceid', 'PRICE.count');
+            .first(
+                'RVBOX.itemcount',
+                'PRICE.priceid',
+                'PRICE.count',
+                'PRICE.barcode',
+            );
 
         if (row === undefined) {
             return undefined;
         }
 
-        const { count, itemcount, priceid } = row;
+        const { count, itemcount, priceid, barcode } = row;
 
         const newCount = count + itemcount * boxCount;
 
         await knex('PRICE')
             .transacting(trx)
             .select({ priceid, endtime: null })
```

Comment:
I think this select can be removed? Nothing is returned.

Edit: Is this actually filtering anything or am I missing something? This type of .select({}) syntax is not found in any Knex docs

Reply:
Actually yes it could be that select is supposed to be where, and instead of where barcode, where priceid should be used to limit the query. I don't know if priceid is unique or not or why endtime: null is there.

Reply:
Could be argued that the current concept of price table is quite unclear and should be refactored to separate price table with only the current price and then another with price history or something like this.

Reply:
> I don't know if priceid is unique (edit: seems to be) or not or why endtime: null is there.

Now looking at this where-ing with the priceid wouldn't work with the current query as it joins to any (previous) price and may not match the last one that's not null. `{ barcode, endtime: null }` always resolves to the correct one but `{ priceid }` only if the query above has `.where({ endtime: null })`

I thought at first the price table was just for recording price history but apparently not. I was wondering why products weren't being sold according to their batch price but this table was for doing exactly that. I think we should implement the way it previously was. Doesn't seem like it would impact any current endpoints apart from /purchase.